components/renku_data_services/message_queue/converters.py

```diff
@@ -98,10 +88,11 @@ class _UserEventConverter:
     @staticmethod
     def to_events(
         user: user_models.UserInfo | user_models.UserWithNamespace | user_models.UserWithNamespaceUpdate,
-        event_type: type[AvroModel] | AmbiguousEvent,
+        event_type: type[AvroModel] | type[events.AmbiguousEvent],
     ) -> list[Event]:
         match event_type:
-            case v2.UserAdded | AmbiguousEvent.INSERT_USER_NAMESPACE if isinstance(user, user_models.UserWithNamespace):
+            case v2.UserAdded | events.InsertUserNamespace:
+                user = cast(user_models.UserWithNamespace, user)
```

Comment:
I know the `if isinstance(...)` guards are ugly but with casting we completely bypass the type checks. So we could be getting something completely random but the value will still be what we cast it to according to the type checker.